src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs

```diff
@@ -220,5 +221,15 @@ private protected void Render(Shape shape, Size? size = null, double scaleX = 1d
 			_mainSvgElement.SetNativeTransform(matrix);
 		}
 
+		internal virtual bool ContainsPoint(Point relativePosition)
```

Comment:
Especially for `Shape` where it not only validates that the point is within the bounds but really validates that pointer is "above drown pixels", I would name it something like `HitTest(Point)` (not sure if could / should remove the old `IsViewHit`).

Also I would put this virtual method directly on `UIElement`, it would also avoid the need to test for `Shape` in the `VisualTreeHelper.HitTest / SerachDownFor...`

Reply:
Yeah changing the name makes sense.

I thought of having it in UIElement, but would the virtual UIElement implementation always return true and keep the renderingBounds check as-is? Or would we move the renderingBounds calculation into it, which may require moving clippingBounds calculation there as well? Or maybe pass renderingBounds as a parameter?

Reply:
> pass renderingBounds as a parameter

I was questioning myself about it, I think it could make sense. (I'm also fine to just return `true` for now and see later to change it if we prefer for now).

Reply:
Does it look better now?